components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
```

Comment:
Broke out to an inportState and outportState to improve the signature.  Is that closer to what you want?